apis/apiextensions/fn/proto/v1beta1/run_function.proto

```diff
@@ -47,6 +47,13 @@ message RunFunctionRequest {
   // Optional input specific to this Function invocation. A JSON representation
   // of the 'input' block of the relevant entry in a Composition's pipeline.
   optional google.protobuf.Struct input = 4;
+
+  // The environment to run the Function with.
+  //
+  // THIS IS AN ALPHA FIELD. Do not use it in production. It is not honored
+  // unless the relevant Crossplane feature flag is enabled, and may be
+  // changed or removed without notice.
```

Comment:
I know I asked for this warning, but maybe we don't need it?

It sounds like we're both pretty confident that we'll need some kind of object that is passed through the Function pipeline as a way for Functions to pass information (that isn't desired state) to each other. So maybe it's fine to commit to this being a beta part of our API without a scary warning. What's "alpha" is more how we might choose to use it (e.g. loading it up with EnvConfigs etc).

Reply:
Yes, I think the in-memory environment is not alpha at all and a pretty useful concept, I agree what's alpha is actually EnvironmentConfig and the way we retrieve that via spec.environment . I'll remove

Reply:
Removed them